### PR TITLE
fix: ignore unnecessary workflows when `**.md` or `ui/**` changes

### PR DESCRIPTION
**Pull request by g1eny0ung**
Signed-off-by: Yue Yang <g1enyy0ung@gmail.com>

### What's changed and how it works?

As the title.

### Related changes

- [ ] This change also requires further updates to the [website](https://github.com/chaos-mesh/website) (e.g. docs)
- [ ] This change also requires further updates to the `UI interface`
- Need to **cheery-pick to release branches**
  - [ ] release-2.4
  - [ ] release-2.3
  - [ ] release-2.2

### Checklist

CHANGELOG

- [ ] I have updated the `CHANGELOG.md`
- [x] I have labeled this PR with "no-need-update-changelog"

Tests

- [ ] Unit test
- [ ] E2E test
- [ ] No code
- [ ] Manual test (add steps below)

Side effects

- [ ] **Breaking backward compatibility**

### DCO

If you find the DCO check fails, please run commands like below (Depends on the actual situations. For example, if the failed commit isn't the most recent) to fix it:

```shell
git commit --amend --signoff
git push --force
```

### CODE REVIEW

**Comment by STRRL** [comment]
I think we do not need to cherrypick into release-2.2 anymore.

**Review by STRRL — APPROVED** [review]
LGTM

**Review by STRRL — CHANGES_REQUESTED** [review]
Oops, I just realized that the Check "E2E Test Passed" is **required** for merging the PR..

**Comment by g1eny0ung** [comment]
> I think we do not need to cherrypick into release-2.2 anymore.

Makes sense. 😌

**Comment by STRRL** [comment]
> Oops, I just realized that the Check "E2E Test Passed" is **required** for merging the PR..

Maybe we should do the same thing like `ci.yaml` and `ci_skip.yaml`.

**Comment by g1eny0ung** [comment]
> Oops, I just realized that the Check "E2E Test Passed" is **required** for merging the PR..

@STRRL We can use the same mechanism as `ci_skip.yaml`. I will update it soon.

File: .github/workflows/integration_test_skip.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+permissions: read-all
+
+jobs:
+  run:
+    name: Integration Test
+    strategy:
+      matrix:
+        arch: [amd64, arm64]
+    runs-on: ${{ fromJson('{"amd64":"ubuntu-latest", "arm64":["self-hosted", "Linux", "ARM64"]}')[matrix.arch] }}
```
Comment (g1eny0ung) [inline .github/workflows/integration_test_skip.yml:19]: @STRRL If we just skip it, shall we just need to write `ubuntu-latest`?

Comment (g1eny0ung) [reply .github/workflows/integration_test_skip.yml]: My mistake. An integration test is not required for merging PRs. I removed this file.

**Comment by g1eny0ung** [comment]
> > Oops, I just realized that the Check "E2E Test Passed" is **required** for merging the PR..
> 
> @STRRL We can use the same mechanism as `ci_skip.yaml`. I will update it soon.

@STRRL After I tried and checked the Actions docs, I found that either `paths` or `dorny/paths-filter@v2` doesn't meet our need because we want **only markdown files to be changed** to not run the `E2E Test` workflow. Both of them will run if at least one other path matches.  :(

So I discover a new way to detect the changes. Using https://github.com/tj-actions/changed-files/ (support `only_changed` detection) and checking whether only markdown files changed. if it is, the rest of the logic is similar to `ci_skip.yml`.

**Review by cwen0 — APPROVED** [review]
LGTM

**Comment by cwen0** [comment]
@STRRL PTAL

**Pull request merged**